OpenLaTeXStudio/EditorModule/src/latexstudio/editor/menu/SaveFile.java

```diff
@@ -43,8 +43,9 @@
     @Override
     public void actionPerformed(ActionEvent e) {                
         String content = etc.getEditorContent();
-        File file = FileChooserService.getSelectedFile("tex", "TeX files", DialogType.SAVE, true);
-        if (file != null) {
+        File file = etc.getCurrentFile();
+        if(file == null) file = FileChooserService.getSelectedFile("tex", "TeX files", DialogType.SAVE, true);
+        if(file != null) {

```

Comment:
In as much as FileChooserService's getSelectedFile method may also return null if the user cancels, I think the second "if" statement is still needed.